src/cmr/concepts/concept.js

```diff
@@ -605,17 +610,21 @@ export default class Concept {
     items.forEach((item) => {
       const normalizedItem = this.normalizeJsonItem(item)
 
-      const { concept_id: conceptId } = normalizedItem
+      const { concept_id: conceptId, revision_id: revisionId } = normalizedItem
 
-      this.setEssentialJsonValues(conceptId, normalizedItem)
+      if (!allRevisions) { this.setEssentialJsonValues(conceptId, normalizedItem) }
 
       jsonKeys.forEach((jsonKey) => {
         const cmrKey = snakeCase(jsonKey)
 
         const { [cmrKey]: keyValue } = normalizedItem
-
         // Snake case the key requested and any children of that key
-        this.setItemValue(conceptId, jsonKey, keyValue)
+
+        let jsonValue = conceptId
+
+        if (allRevisions) { jsonValue = `Revision${revisionId}` }
```

Comment:
I am a little confused on whats happening here. I could probably use a quick refresher on this.